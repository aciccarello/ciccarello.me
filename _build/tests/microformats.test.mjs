import { mf2tojf2 } from '@paulrobertlloyd/mf2tojf2';
import fs from 'node:fs/promises';
import { mf2 } from 'microformats-parser';

/** @typedef {import('../../_data/test.json')} testData */

describe('microformats', () => {
  /** @type {testData['posts']} */
  let posts;
  /**
   * Test posts to test.
   *
   * Needed because jest needs test names synchronously (and testData is async)
   * @type {Array<keyof testData['posts']>}
   */
  const testPostNames = [
    'article',
    'recipe',
    'note',
    'photo',
    'like',
    'photoLike',
    'reply',
    'replyToCheckin',
  ];

  beforeAll(async () => {
    const testData = JSON.parse(await fs.readFile('_data/test.json', 'utf8'));
    posts = testData.posts;
  });

  it('should have an h-card on the homepage', async () => {
    const homepage = await getPage('/');

    expect(jf2(homepage)).toMatchInlineSnapshot(`
      {
        "email": "mailto:anthony@ciccarello.me",
        "name": "Anthony Ciccarello",
        "note": "I'm a software engineer living in Southern California building cool things using JavaScript and other web technologies. I enjoy traveling to other countries and spending time in nature.",
        "org": {
          "name": "ICD",
          "type": "card",
          "url": "https://icdportal.com/",
        },
        "photo": "https://secure.gravatar.com/avatar/17d306899b5f20953440eca1d65d34e0?s=512",
        "type": "card",
        "url": [
          "https://www.ciccarello.me/",
          "acct:anthony@ciccarello.me",
          "https://www.ciccarello.me/subscribe/#follow-on-mastodon-or-other-fediverse-networks",
        ],
      }
    `);
  });

  it('should be parsable on the resume', async () => {
    const homepage = await getPage('/resume/');

    expect(jf2(homepage)).toMatchSnapshot();
  });

  it('should have h-entries on the posts page', async () => {
    const page = await getPage('/posts/testPosts/');

    expect(jf2(page)).toMatchSnapshot();
  });

  testPostNames.forEach((description) => {
    it('should have a(n) ' + description, async () => {
      const page = await getPage(posts[description]);

      expect(jf2Main(page)).toMatchSnapshot();
    });
  });
});

async function getPage(path) {
  return fs.readFile(`_site${path}index.html`, 'utf-8');
}

function jf2(page) {
  return mf2tojf2(mf2(page, { baseUrl: 'https://www.ciccarello.me' }));
}

function jf2Main(page) {
  const complete = jf2(page);
  return complete.children ? complete.children[0] : complete;
}
