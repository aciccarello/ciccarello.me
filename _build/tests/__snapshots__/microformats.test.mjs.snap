// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`microformats should be parsable on the resume 1`] = `
{
  "education": {
    "end": "2014",
    "location": {
      "org": "Cedarville University",
      "type": "card",
    },
    "name": "Bachelor of Science (BS), Computer Engineering",
    "start": "2010",
    "summary": "Minor in Mathematics, and Computer Science.",
    "type": "event",
  },
  "experience": {
    "children": [
      {
        "location": {
          "children": [
            {
              "org": "ICD",
              "type": "card",
              "url": "https://icdportal.com/",
            },
            "Remote",
          ],
        },
        "name": "Senior Software Engineer",
        "skill": [
          "React",
          "JavaScript",
          "Node.js",
          "Git",
          "Agile",
          "Scrum",
          "Java",
          "Groovy",
          "Spring",
          "Oracle SQL",
          "Automated Testing",
        ],
        "start": "2023-05",
        "summary": "Full-stack development of ICD Portal investment platform.

						
							
								
									Helped team revamp Agile processes and consulted on setup of Jira project management.
								
							
								
									Led migration of existing codebase to TypeScript.
								
							
								
									Created web app to view automation testing result logs.",
        "type": "event",
      },
      {
        "end": "2023-04",
        "location": {
          "children": [
            {
              "org": "SitePen",
              "type": "card",
              "url": "https://www.sitepen.com/",
            },
            "Remote",
          ],
        },
        "name": "Software Engineer",
        "skill": [
          "Angular",
          "React",
          "JavaScript",
          "TypeScript",
          "AWS Lambda",
          "DynamoDB",
          "Elastic Beanstalk",
          "Node.js",
          "SCSS",
          "Git",
          "Agile",
          "Scrum",
          "Java",
          "Spring",
          "REST",
          "Express.js",
          "Nest.js",
          "Team Leadership",
          "Coaching",
          "CI/CD",
          "Jenkins",
          "Atlassian",
        ],
        "start": "2018-08",
        "summary": "Full-stack software development and consulting work on cross-functional teams for multiple clients. Primarily focused on JavaScript/TypeScript development.

						
							
								
									Acted as team lead on Angular application development for a financial services corporation.
								
							
								
									Implemented main page redesign with team and developed interactive React components for service cross-selling in major online accounting application
								
							
								
									Worked on and with AWS Lambda, DynamoDB, and Elastic Beanstalk back-ends using Express.js and Java Spring.
								
							
								
									Lead test automation implementation and 6-month junior developer coaching program.
								
							
							
								
									Helped deliver a business management platform using Dojo 2 and a Nest.js backend",
        "type": "event",
      },
      {
        "end": "2018-08",
        "location": {
          "children": [
            {
              "org": "DISH Network",
              "type": "card",
              "url": "https://www.dish.com/",
            },
            "Denver, CO",
          ],
        },
        "name": "Software Engineer",
        "skill": [
          "Java",
          "Spring",
          "REST",
          "MongoDB",
          "TDD",
          "Extreme Programming",
          "Agile",
          "Rally",
        ],
        "start": "2018-05",
        "summary": "Contract position with central customer data API team.

						
							
								
									Developed Java REST API microservices using Spring Boot in XP/TDD environment
								
							
								
									Helped inform API and architectural designs for development team
								
							
							
								
									Started regular team discussion on how to improve team processes and designs",
        "type": "event",
      },
      {
        "end": "2018-04",
        "location": {
          "children": [
            {
              "org": "Teradata",
              "type": "card",
              "url": "https://www.teradata.com/",
            },
            "Dayton, OH/Remote",
          ],
        },
        "name": "Software/Sys Eng Anlyst, IT Services",
        "skill": [
          "AngularJS",
          "ag-grid",
          "Java",
          "Spring",
          "Bootstrap",
          "Oracle SQL",
          "JSP",
          "Classic ASP",
        ],
        "start": "2014-06",
        "summary": "Developed and maintained internal HR applications for 10,000 employee company. Applications I helped maintain included payroll, sales compensation, and benefits tools.

						
							
								
									Designed and lead new AngularJS/ag-grid web application deployment/testing
								
							
								
									Developed Java Spring and Oracle SQL database backend
								
							
							
								
									Supported legacy JSP and Classic ASP pages",
        "type": "event",
      },
      {
        "end": "2021-01",
        "location": {
          "children": [
            {
              "org": "Covenant Kitchens & Baths",
              "type": "card",
              "url": "https://www.covenantkitchens.com/",
            },
            "Westbrook, CT/Remote",
          ],
        },
        "name": "Web Developer/Systems Technician/CAD Draftsman",
        "skill": [
          "Web design",
          "System Administration",
          "SEO",
          "GSuite",
          "AutoCAD",
          "SketchUp",
        ],
        "start": "2011-05",
        "summary": "Range of technical work for Kitchen & Bath design firm. CAD Drafting in-person and web administration later as freelance.

						
							
								
									Completed website revamp and performed G-Suite administration and SEO optimization
								
							
							
								
									Produced 2D & 3D drawings in AutoCAD and SketchUp
								
							
								
									Transferred email system to Google Apps
								
							
								
									Provided website and technical support on a contract basis",
        "type": "event",
      },
      {
        "end": "2013-08",
        "location": {
          "children": [
            {
              "name": "Alstom Power",
              "type": "card",
            },
            "Hartford, CT",
          ],
        },
        "name": "Information Systems Coordinator/Business Intelligence Intern",
        "skill": [
          "SAP",
          "BusinessObjects",
          "Hyperion",
          "Excel Macros",
          "VBScript",
        ],
        "start": "2012-07",
        "summary": "Summer internship with the business intelligence team of a multinational company.

						
							
								
									Automated manual data entry into SAP using VBScript, speeding up migration",
        "type": "event",
      },
    ],
  },
  "name": {
    "name": "Anthony Ciccarello",
    "type": "card",
    "url": "https://www.ciccarello.me/",
  },
  "skill": [
    "JavaScript",
    "TypeScript",
    "Angular",
    "React",
    "Java",
    "HTML5",
    "CSS",
    "Node.js",
    "AWS",
    "Maven",
    "Git",
    "Automated Testing",
  ],
  "summary": "Software Engineer experienced in modern JavaScript front-end frameworks and back-end REST APIs.
		Knowledgeable in development best practices, automated testing, and agile methodologies.
		Seeking a remote front end or full stack engineer position building high-quality websites regardless of technology.",
  "type": "resume",
}
`;

exports[`microformats should have a(n) article 1`] = `
{
  "author": {
    "name": "Anthony Ciccarello",
    "photo": "https://secure.gravatar.com/avatar/17d306899b5f20953440eca1d65d34e0?s=512",
    "summary": "I'm a software engineer living in Southern California building cool things using JavaScript and other web technologies. I enjoy traveling to other countries and spending time in nature.",
    "type": "card",
    "url": "https://www.ciccarello.me/",
  },
  "bridgy-fed": "https://fed.brid.gy/",
  "category": [
    "travel",
    "Senegal",
    "Liberia",
  ],
  "content": {
    "html": "<p>The past two days have destroyed my sense of time. Between changing time zones
twice, flying through the night, flying for more than 6 hours at a time,
having periodic naps, and standing in lines for long stretches, I have stopped
keeping a close eye on time. I guess that is a fitting entrance to Africa. On
Tuesday I was counting every minute making sure I worked enough hours for the
week and rushing through getting out the door to make our plane on time. This
afternoon we didn’t have to manage ant firm times for events to happen. Our
plan was to fly directly from Brussels to Monrovia, Liberia. God apparently
had other plans. After about four hours into our “last flight” to Liberia the
pilots decided it was necessary to divert the flight to Senegal, several
countries up the west coast of Africa. We would find out later that some
unexpected bad weather had caused a crack in the windshield of the airplane
and the Dakar airport was better prepared to make the repairs.</p>
<figure><img src="https://www.ciccarello.me/assets/img/senegal-leaving-airport.jpg" alt="Crowd of people in a parking lot with lots of luggage with a white bus" class="post-img--float"><figcaption>Waiting outside the airport for a bus to the&nbsp;hotel</figcaption></figure>
<p>We followed the mass of other displaced travelers out of the airport to where
bus after bus came to shuttle people and their luggage to our hotel. But this
long wait provided us with an opportunity to connect with the other people on
our flight to Monrovia including native Liberians and humanitarian/missions
workers. This first look at Africa was a great introduction to the differences
in culture that we may have to face. I enjoyed looking at the way people
dressed, how they interacted with each other, and how the English speaking
Liberians interacted with the primarily French speaking Senegal workers. When
our boxes finally got loaded on a bus and we packed inside, we got a small
tour of the city on the way to our hotel. I was reminded of many scenes from
Bekah and my trip to Kosovo in eastern Europe last year. Things like the
concrete buildings, the mix of businesses, and people standing around. But
there were also significant differences. Dakar Senegal had more poverty, more
English, and more social/open people than Mitrovica, Kosovo. It was great to
explore this new world.</p>
<figure><img src="https://www.ciccarello.me/assets/img/senegal-hotel-night.jpg" alt="View of a pool surrounded by palm trees at night"><figcaption>Nighttime view from our hotel room overlooking the ocean and pool</figcaption></figure>
<p>As we arrived at the hotel, we were pleasantly surprised to find our temporary
housing was a upscale place. We were able to get an ocean view room and meal
all provided by Brussels Airlines. After so much traveling, it was a blessing
to be able to relax (and take a shower).</p>
<figure><img src="https://www.ciccarello.me/assets/img/senegal-hotel-day.jpg" alt="View of the ocean and a pool surrounded by palm trees from a hotel balcony"><figcaption>Daytime view from&nbsp;hotel</figcaption></figure>
<p>Things are still up in the air for our real final flight to Liberia but for
being stick for a while, we are pretty well off. We will also be looking to
see how God uses this delay. Update: Our flight to Monrovia has been set for
Sunday meaning we will be in Senegal another day.</p>
<hr>
<p><em>Originally published at <a rel="syndication" class="u-syndication" href="https://liveloveliberia.blogspot.com/2015/07/getting-introduction-to-africa.html">liveloveliberia.blogspot.com</a> on July 24, 2015.</em></p>",
    "text": "The past two days have destroyed my sense of time. Between changing time zones
twice, flying through the night, flying for more than 6 hours at a time,
having periodic naps, and standing in lines for long stretches, I have stopped
keeping a close eye on time. I guess that is a fitting entrance to Africa. On
Tuesday I was counting every minute making sure I worked enough hours for the
week and rushing through getting out the door to make our plane on time. This
afternoon we didn’t have to manage ant firm times for events to happen. Our
plan was to fly directly from Brussels to Monrovia, Liberia. God apparently
had other plans. After about four hours into our “last flight” to Liberia the
pilots decided it was necessary to divert the flight to Senegal, several
countries up the west coast of Africa. We would find out later that some
unexpected bad weather had caused a crack in the windshield of the airplane
and the Dakar airport was better prepared to make the repairs.
 Crowd of people in a parking lot with lots of luggage with a white bus Waiting outside the airport for a bus to the hotel
We followed the mass of other displaced travelers out of the airport to where
bus after bus came to shuttle people and their luggage to our hotel. But this
long wait provided us with an opportunity to connect with the other people on
our flight to Monrovia including native Liberians and humanitarian/missions
workers. This first look at Africa was a great introduction to the differences
in culture that we may have to face. I enjoyed looking at the way people
dressed, how they interacted with each other, and how the English speaking
Liberians interacted with the primarily French speaking Senegal workers. When
our boxes finally got loaded on a bus and we packed inside, we got a small
tour of the city on the way to our hotel. I was reminded of many scenes from
Bekah and my trip to Kosovo in eastern Europe last year. Things like the
concrete buildings, the mix of businesses, and people standing around. But
there were also significant differences. Dakar Senegal had more poverty, more
English, and more social/open people than Mitrovica, Kosovo. It was great to
explore this new world.
 View of a pool surrounded by palm trees at night Nighttime view from our hotel room overlooking the ocean and pool
As we arrived at the hotel, we were pleasantly surprised to find our temporary
housing was a upscale place. We were able to get an ocean view room and meal
all provided by Brussels Airlines. After so much traveling, it was a blessing
to be able to relax (and take a shower).
 View of the ocean and a pool surrounded by palm trees from a hotel balcony Daytime view from hotel
Things are still up in the air for our real final flight to Liberia but for
being stick for a while, we are pretty well off. We will also be looking to
see how God uses this delay. Update: Our flight to Monrovia has been set for
Sunday meaning we will be in Senegal another day.

Originally published at liveloveliberia.blogspot.com on July 24, 2015.",
  },
  "featured": [
    {
      "alt": "large monument of a woman and a man holding a small child with people on steps leading up to the monument",
      "url": "https://www.ciccarello.me/assets/img/senegal-african-renaissance-monument.jpg",
    },
  ],
  "name": "Getting An Introduction to Africa",
  "published": "2015-07-24",
  "summary": "Detour to Senegal",
  "syndication": [
    "https://liveloveliberia.blogspot.com/2015/07/getting-introduction-to-africa.html",
    "https://medium.com/@aciccarello/getting-an-introduction-to-africa-91d11e5914e7",
  ],
  "type": "entry",
  "url": "https://www.ciccarello.me/blog/2015/07/24/getting-an-introduction-to-africa/",
}
`;

exports[`microformats should have a(n) like 1`] = `
{
  "author": {
    "name": "Anthony Ciccarello",
    "photo": "https://secure.gravatar.com/avatar/17d306899b5f20953440eca1d65d34e0?s=512",
    "summary": "I'm a software engineer living in Southern California building cool things using JavaScript and other web technologies. I enjoy traveling to other countries and spending time in nature.",
    "type": "card",
    "url": "https://www.ciccarello.me/",
  },
  "bridgy-fed": "https://fed.brid.gy/",
  "content": {
    "html": "",
  },
  "like-of": {
    "author": {
      "name": "James' Coffee Blog ☕",
      "type": "card",
      "url": "https://jamesg.blog",
    },
    "content": "I have previously taken long breaks from coding outside of work, some of which have lasted months. I remember last time there was a voice in my brain that said "this is enough" and I decided to take a step back. Building projects for work is one thing; I have support and motivation and I am contributing toward a goal that a whole team is working on. But when I work on personal projects, sometimes I get stuck working on things that don't really matter and that I spe…",
    "type": "cite",
    "url": "https://jamesg.blog/2023/01/03/coding-mental-health/",
  },
  "published": "2023-01-27T07:38:26Z",
  "type": "entry",
  "url": "https://www.ciccarello.me/posts/2023/01/27/jamesg-blog-like/",
}
`;

exports[`microformats should have a(n) multipleUpdates 1`] = `
{
  "author": {
    "name": "Anthony Ciccarello",
    "photo": "https://secure.gravatar.com/avatar/17d306899b5f20953440eca1d65d34e0?s=512",
    "summary": "I'm a software engineer living in Southern California building cool things using JavaScript and other web technologies. I enjoy traveling to other countries and spending time in nature.",
    "type": "card",
    "url": "https://www.ciccarello.me/",
  },
  "bridgy-fed": "https://fed.brid.gy/",
  "content": {
    "html": "<style>
  .dt-updated {
    color: red;
  }
</style>
<p>This post is for testing how updates are handled.
Typically we’ll want the updates to be in order but that’s not guaranteed.
So we want to check for the date matching the calculated <code>lastUpdated</code> value.
The most recent update will get the microformats dt-updated class.</p>
<p>The lastUpdated value is: 2023-10-23T06:43:26Z</p>",
    "text": "This post is for testing how updates are handled.
Typically we’ll want the updates to be in order but that’s not guaranteed.
So we want to check for the date matching the calculated lastUpdated value.
The most recent update will get the microformats dt-updated class.
The lastUpdated value is: 2023-10-23T06:43:26Z",
  },
  "published": "2023-10-23T05:46:06Z",
  "type": "entry",
  "updated": "2023-10-23T06:43:26Z",
  "url": "https://www.ciccarello.me/posts/2023/10/23/updates-test/",
}
`;

exports[`microformats should have a(n) note 1`] = `
{
  "author": {
    "name": "Anthony Ciccarello",
    "photo": "https://secure.gravatar.com/avatar/17d306899b5f20953440eca1d65d34e0?s=512",
    "summary": "I'm a software engineer living in Southern California building cool things using JavaScript and other web technologies. I enjoy traveling to other countries and spending time in nature.",
    "type": "card",
    "url": "https://www.ciccarello.me/",
  },
  "bridgy-fed": "https://fed.brid.gy/",
  "category": [
    "IndieWeb",
    "site changes",
  ],
  "content": {
    "html": "<p>I’ve added a new type of post to my site. This should open a whole world of possibilities integrating with <a href="http://indieweb.org">indieweb.org</a> standards!</p>",
    "text": "I’ve added a new type of post to my site. This should open a whole world of possibilities integrating with indieweb.org standards!",
  },
  "published": "2022-03-30T05:16:18Z",
  "type": "entry",
  "url": "https://www.ciccarello.me/posts/2022/03/30/first-note/",
}
`;

exports[`microformats should have a(n) photo 1`] = `
{
  "author": {
    "name": "Anthony Ciccarello",
    "photo": "https://secure.gravatar.com/avatar/17d306899b5f20953440eca1d65d34e0?s=512",
    "summary": "I'm a software engineer living in Southern California building cool things using JavaScript and other web technologies. I enjoy traveling to other countries and spending time in nature.",
    "type": "card",
    "url": "https://www.ciccarello.me/",
  },
  "bridgy-fed": "https://fed.brid.gy/",
  "category": [
    "nature",
    "ocean",
    "site changes",
  ],
  "content": {
    "html": "<p>Photography has been a fun hobby to pick up. Though I may backfill older posts I wanted to start with something fresh. I’m looking forward to adding more content to my site.</p>",
    "text": "Photography has been a fun hobby to pick up. Though I may backfill older posts I wanted to start with something fresh. I’m looking forward to adding more content to my site.",
  },
  "featured": [
    {
      "alt": "Flowering bush in front of ocean coast.",
      "url": "https://www.ciccarello.me/assets/img/2021-nor-cal-coast.jpg",
    },
  ],
  "photo": [
    {
      "alt": "Flowering bush in front of ocean coast.",
      "url": "https://www.ciccarello.me/assets/img/2021-nor-cal-coast.jpg",
    },
  ],
  "published": "2021-06-21",
  "type": "entry",
  "url": "https://www.ciccarello.me/photos/2021/06/21/initial-photo/",
}
`;

exports[`microformats should have a(n) photoLike 1`] = `
{
  "author": {
    "name": "Anthony Ciccarello",
    "photo": "https://secure.gravatar.com/avatar/17d306899b5f20953440eca1d65d34e0?s=512",
    "summary": "I'm a software engineer living in Southern California building cool things using JavaScript and other web technologies. I enjoy traveling to other countries and spending time in nature.",
    "type": "card",
    "url": "https://www.ciccarello.me/",
  },
  "bridgy-fed": "https://fed.brid.gy/",
  "content": {
    "html": "",
  },
  "like-of": {
    "author": {
      "name": "Paul Robert Lloyd",
      "type": "card",
      "url": "https://paulrobertlloyd.com/",
    },
    "content": "Summer sunshine.",
    "photo": [
      {
        "alt": "Rays of sunlight poke through trees in a park as it sets in the distance, giving everything a vivid colour.",
        "url": "https://paulrobertlloyd.com/media/2023/157/p1/1.jpg?tr=w-960",
      },
    ],
    "type": "cite",
    "url": "https://paulrobertlloyd.com/2023/157/p1/",
  },
  "published": "2023-07-30T20:07:10Z",
  "type": "entry",
  "url": "https://www.ciccarello.me/posts/2023/07/30/summer-sunshine-like/",
}
`;

exports[`microformats should have a(n) recipe 1`] = `
{
  "author": {
    "name": "Anthony Ciccarello",
    "photo": "https://secure.gravatar.com/avatar/17d306899b5f20953440eca1d65d34e0?s=512",
    "summary": "I'm a software engineer who enjoys collecting recipes I've made and preserving family favorites. My cooking style is laid back and flexible so some recipes may not be exact.",
    "type": "card",
    "url": "https://www.ciccarello.me/",
  },
  "category": [
    "drink",
    "holiday",
    "family recipe",
  ],
  "content": {
    "html": "<p>This simple recipe is family favorite over the holidays.</p>
<h2 id="ingredients" tabindex="-1">Ingredients</h2>
<ul>
<li class="p-ingredient ingredient">Ginger ale</li>
<li class="p-ingredient ingredient">Eggnog</li>
<li class="p-ingredient ingredient">Vanilla Ice Cream</li>
<li class="p-ingredient ingredient">Cinnamon</li>
</ul>
<h2 id="directions" tabindex="-1">Directions</h2>
<ol class="e-instructions instructions">
<li class="p-instruction instruction">Combine ginger ale and eggnog in a large container</li>
<li class="p-instruction instruction">Add scoops of vanilla ice cream</li>
<li class="p-instruction instruction">Sprinkle cinnamon on top</li>
</ol>",
    "text": "This simple recipe is family favorite over the holidays.
Ingredients

Ginger ale
Eggnog
Vanilla Ice Cream
Cinnamon

Directions

Combine ginger ale and eggnog in a large container
Add scoops of vanilla ice cream
Sprinkle cinnamon on top",
  },
  "featured": [
    {
      "alt": "short glass with creamy drink and cinnamon next to coffee cake.",
      "url": "https://www.ciccarello.me/assets/img/eggnog-punch.jpg",
    },
  ],
  "ingredient": [
    "Ginger ale",
    "Eggnog",
    "Vanilla Ice Cream",
    "Cinnamon",
  ],
  "instruction": [
    "Combine ginger ale and eggnog in a large container",
    "Add scoops of vanilla ice cream",
    "Sprinkle cinnamon on top",
  ],
  "instructions": {
    "html": "<li class="p-instruction instruction">Combine ginger ale and eggnog in a large container</li>
<li class="p-instruction instruction">Add scoops of vanilla ice cream</li>
<li class="p-instruction instruction">Sprinkle cinnamon on top</li>",
    "text": "Combine ginger ale and eggnog in a large container
Add scoops of vanilla ice cream
Sprinkle cinnamon on top",
  },
  "name": "Eggnog Punch",
  "photo": [
    {
      "alt": "short glass with creamy drink and cinnamon next to coffee cake.",
      "url": "https://www.ciccarello.me/assets/img/eggnog-punch.jpg",
    },
  ],
  "published": "2022-01-01T04:14:11Z",
  "type": "recipe",
  "url": "https://www.ciccarello.me/recipes/2022/01/01/eggnog-punch/",
}
`;

exports[`microformats should have a(n) reply 1`] = `
{
  "author": {
    "name": "Anthony Ciccarello",
    "photo": "https://secure.gravatar.com/avatar/17d306899b5f20953440eca1d65d34e0?s=512",
    "summary": "I'm a software engineer living in Southern California building cool things using JavaScript and other web technologies. I enjoy traveling to other countries and spending time in nature.",
    "type": "card",
    "url": "https://www.ciccarello.me/",
  },
  "bridgy-fed": "https://fed.brid.gy/",
  "category": "IndieWeb",
  "content": {
    "html": "<p>I know there’s an <a href="https://github.com/snarfed/bridgy-fed/issues/272">open issue</a> to add documentation but if it’s helpful you can look at the <a href="https://github.com/aciccarello/ciccarello.me/compare/d5f25dec5a441fb4f6783facd54e88de30250c0f...61457954adc86d34a67080313cabf24f11ac4eba">relevant commits</a> <a href="https://github.com/aciccarello/ciccarello.me/commit/50a67193255fb81377d77a790b830907469fcc44">on my site</a>. The piece missing from the docs is adding a link with a class of “u-url” and an href with your username in the format <code>acct:anthony@ciccarello.me</code> to your homepage/h-card.</p>",
    "text": "I know there’s an open issue to add documentation but if it’s helpful you can look at the relevant commits on my site. The piece missing from the docs is adding a link with a class of “u-url” and an href with your username in the format acct:anthony@ciccarello.me to your homepage/h-card.",
  },
  "in-reply-to": {
    "author": {
      "name": "Abhinav",
      "type": "card",
      "url": "https://types.pl/@abnv",
    },
    "content": "@aciccarello It would be great if you can publish the instructions for the same. I've tried following the instructions on Bridgy Fed website but they didn't work for me.",
    "type": "cite",
    "url": "https://types.pl/@abnv/109360439631118847",
  },
  "published": "2022-11-17T18:41:14Z",
  "type": "entry",
  "url": "https://www.ciccarello.me/posts/2022/11/17/bridgy-fed-instructions/",
}
`;

exports[`microformats should have a(n) replyToCheckin 1`] = `
{
  "author": {
    "name": "Anthony Ciccarello",
    "photo": "https://secure.gravatar.com/avatar/17d306899b5f20953440eca1d65d34e0?s=512",
    "summary": "I'm a software engineer living in Southern California building cool things using JavaScript and other web technologies. I enjoy traveling to other countries and spending time in nature.",
    "type": "card",
    "url": "https://www.ciccarello.me/",
  },
  "bridgy-fed": "https://fed.brid.gy/",
  "content": {
    "html": "<p>Don’t say I never sent you a webmention.</p>",
    "text": "Don’t say I never sent you a webmention.",
  },
  "in-reply-to": {
    "author": {
      "name": "David Shanske",
      "type": "card",
      "url": "https://david.shanske.com/",
    },
    "content": "Checked into Aspire Lounge San Diego",
    "type": "cite",
    "url": "https://david.shanske.com/2022/11/12/5771/",
  },
  "published": "2023-01-26T06:04:30Z",
  "type": "entry",
  "url": "https://www.ciccarello.me/posts/2023/01/26/a-webmention-for-david/",
}
`;

exports[`microformats should have an h-card on the homepage 1`] = `
{
  "children": [
    {
      "children": [
        "Changing content removed",
      ],
      "name": "Latest Posts",
      "type": "feed",
    },
  ],
  "email": "mailto:anthony@ciccarello.me",
  "name": "Anthony Ciccarello",
  "note": "I'm a software engineer living in Southern California building cool things using JavaScript and other web technologies. I enjoy traveling to other countries and spending time in nature.",
  "org": {
    "name": "ICD",
    "type": "card",
    "url": "https://icdportal.com/",
  },
  "photo": "https://secure.gravatar.com/avatar/17d306899b5f20953440eca1d65d34e0?s=512",
  "type": "card",
  "url": [
    "https://www.ciccarello.me/",
    "acct:anthony@ciccarello.me",
  ],
}
`;

exports[`microformats should have h-entries on the posts page 1`] = `
{
  "children": [
    {
      "author": "https://www.ciccarello.me/",
      "category": [
        "travel",
        "Senegal",
        "Liberia",
      ],
      "name": "Getting An Introduction to Africa",
      "published": "2015-07-24",
      "summary": "Detour to Senegal",
      "type": "entry",
      "url": "https://www.ciccarello.me/blog/2015/07/24/getting-an-introduction-to-africa/",
    },
    {
      "author": "https://www.ciccarello.me/",
      "category": [
        "nature",
        "ocean",
        "site changes",
      ],
      "published": "2021-06-21",
      "summary": "Photography has been a fun hobby to pick up. Though I may backfill older posts I wanted to start with something fresh. I’m looking forward to adding more content to my site.",
      "type": "entry",
      "url": "https://www.ciccarello.me/photos/2021/06/21/initial-photo/",
    },
    {
      "author": "https://www.ciccarello.me/",
      "category": [
        "drink",
        "holiday",
        "family recipe",
      ],
      "name": "Eggnog Punch",
      "published": "2022-01-01T04:14:11Z",
      "summary": "This simple recipe is family favorite over the holidays. Ingredients Ginger ale Eggnog Vanilla Ice Cream Cinnamon Directions Combine ginger ale and eggnog in a large container Add scoops of vanilla ice cream Sprinkle cinnamon on top",
      "type": "recipe",
      "url": "https://www.ciccarello.me/recipes/2022/01/01/eggnog-punch/",
    },
    {
      "author": "https://www.ciccarello.me/",
      "category": [
        "IndieWeb",
        "site changes",
      ],
      "published": "2022-03-30T05:16:18Z",
      "summary": "I’ve added a new type of post to my site. This should open a whole world of possibilities integrating with indieweb.org standards!",
      "type": "entry",
      "url": "https://www.ciccarello.me/posts/2022/03/30/first-note/",
    },
    {
      "author": "https://www.ciccarello.me/",
      "category": "IndieWeb",
      "published": "2022-11-17T18:41:14Z",
      "summary": "I know there’s an open issue to add documentation but if it’s helpful you can look at the relevant commits on my site. The piece missing from the docs is adding a link with a class of “u-url” and an href with your username in the format acct:anthony@ciccarello.me to your...",
      "type": "entry",
      "url": "https://www.ciccarello.me/posts/2022/11/17/bridgy-fed-instructions/",
    },
    {
      "author": "https://www.ciccarello.me/",
      "published": "2023-01-26T06:04:30Z",
      "summary": "Don’t say I never sent you a webmention.",
      "type": "entry",
      "url": "https://www.ciccarello.me/posts/2023/01/26/a-webmention-for-david/",
    },
    {
      "author": "https://www.ciccarello.me/",
      "published": "2023-01-27T07:38:26Z",
      "summary": "Liked “Taking a break from personal projects: Mental health and coding” by James' Coffee Blog ☕",
      "type": "entry",
      "url": "https://www.ciccarello.me/posts/2023/01/27/jamesg-blog-like/",
    },
    {
      "author": "https://www.ciccarello.me/",
      "published": "2023-07-30T20:07:10Z",
      "summary": "Liked a photo by Paul Robert Lloyd:
							“Summer sunshine.”",
      "type": "entry",
      "url": "https://www.ciccarello.me/posts/2023/07/30/summer-sunshine-like/",
    },
  ],
  "name": "Test post list",
  "type": "feed",
}
`;
